#167729782-docker-setup

okabamac opened the pull request:
## What does this PR do?
- Setup dockerfiles and dockercompose scripts

## Description of Task to be completed?
- Create dockerfile for building images
- Create dockercompose scripts to enable running docker containers easy
- Delete .env.sample file because of duplication
- Write scripts in package.json to make running commands easy

## How should this be manually tested?
pull or fetch this branch
Run the following commands after you cd into the directory
``` docker-compose -f docker-compose.dev.yml up --build ``` 
To run it inside dockers(ensure you have docker and docker-compose installed
Any background context you want to provide?
No

## What are the relevant pivotal tracker stories?
[#167729782](https://www.pivotaltracker.com/story/show/167729782)

* Screenshots (if appropriate)

Questions 
No

okabamac committed:
ch(docker): docker-configuration
- Add dockerfile script for multistage build
- Add docker-compose script to ensure easy build and service running
- Delete .env.sample file
- Add .travis.yml file to run docker in travis
- Rewrite test/dev commands in package.json
[Finishes #167729782]

okabamac committed:
ch(docker-setup): configure docker to build images
- Add dockerfile to enable multistage build
- Add docker compose scripts to stopping and running container services easily
- Add travis scripts to ensure travis runs test inside containers
- Write scripts in package.json to handle simple commands
[Finishes #167729782]

okabamac committed:
Delete .travis.yml to fix errored build

okafor-chidimma reviewed (approved):
No merge conflicts...Approved for merge

okabamac committed:
Resolve merge conflict
Add a service for postgres test
Update package.json to enable browser debugging

okabamac committed:
Remove parenthesis around single arrow function to fix hounding

davidfrank96 commented on docker-compose.prod.yml:
```diff
@@ -0,0 +1,27 @@
+    networks: 
+      - webnet
+networks: 
+  webnet:
```
This red symbol means that after writing code ..you should always add space at the end... so try to implement that moving forward

okabamac replied:
OK. Thanks a lot, I probably forgot to run lint at the end of the last commit.

davidfrank96 reviewed (approved):
Looks good bruh, just check out the little comment below
